src/app/Http/Controllers/UserCrudController.php

```diff
@@ -111,49 +114,40 @@ public function setup()
      */
     public function store(StoreRequest $request)
     {
-        $this->crud->hasAccessOrFail('create');
-
-        // insert item in the db
-        if ($request->input('password')) {
-            $item = $this->crud->create(\Request::except(['redirect_after_save']));
+        $this->handlePasswordInput($request);
 
-            // now bcrypt the password
-            $item->password = bcrypt($request->input('password'));
-            $item->save();
-        } else {
-            $item = $this->crud->create(\Request::except(['redirect_after_save', 'password']));
-        }
-
-        // show a success message
-        \Alert::success(trans('backpack::crud.insert_success'))->flash();
-
-        // save the redirect choice for next time
-        $this->setSaveAction();
-
-        return $this->performSaveAction($item->getKey());
+        return parent::storeCrud($request);
     }
 
+    /**
+     * Update the specified resource in the database.
+     *
+     * @param UpdateRequest $request - type injection used for validation using Requests
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
     public function update(UpdateRequest $request)
     {
-        //encrypt password and set it to request
-        $this->crud->hasAccessOrFail('update');
+        $this->handlePasswordInput($request);
 
-        $dataToUpdate = \Request::except(['redirect_after_save', 'password']);
+        return parent::updateCrud($request);
+    }
+
+    /**
+     * Handle password input fields.
+     *
+     * @param CrudRequest $request
+     */
+    protected function handlePasswordInput(CrudRequest $request)
+    {
+        // Remove fields not present on the user.
+        $request->request->remove('password_confirmation');
 
-        //encrypt password
+        // Encrypt password if specified.
         if ($request->input('password')) {
-            $dataToUpdate['password'] = bcrypt($request->input('password'));
+            $request->request->set('password', bcrypt($request->input('password')));
```

Comment:
I wonder is this should be `Hash::make($request->input('password)`?

The Hash façade seems to have existed since 2013.

Pros:

1. If upstream meddle with passwords this is probably the standard method they'll use.

Cons:

1. It's a little orthogonal to this particular change